packages/compiler/src/core/checker.ts

```diff
@@ -2678,6 +2678,7 @@ export function createChecker(program: Program): Checker {
           }
           for (const prop of node.properties) {
             if (prop.kind === SyntaxKind.ModelSpreadProperty) {
+              checkForSelfReference(prop, node);
```

Comment:
Ok other thing I am not sure we should do this "check" in the binding phase might be better in the `checkModelProperties`

Reply:
Makes sense to me.

Reply:
Relocating the logic made it much more concise. Thanks for the pointer!